resources/sip11/ra/src/main/java/org/mobicents/slee/resource/sip11/wrappers/DialogWrapper.java

```diff
@@ -501,9 +501,17 @@ public void delete() {
 		}
 		
 		final DialogState currentState = wrappedDialog != null ? wrappedDialog.getState() : null;
+		boolean isServer = wrappedDialog != null ? wrappedDialog.isServer() : false; // is false ok, here?
 		boolean stackDoesNotFiresDialogTerminatedEvent = !isEnding()
-			&& !wrappedDialog.isServer() && (currentState == null || currentState == DialogState.TERMINATED);
-		wrappedDialog.delete();
+			&& !isServer() && (currentState == null || currentState == DialogState.TERMINATED);
```

Comment:
shouldnt we use the previously defined "isServer" local var, instead of calling isServer() method which will result in NPE again?